zk_toolbox/crates/zk_supervisor/src/commands/lint.rs

```diff
@@ -0,0 +1,194 @@
+    "target",
+    "node_modules",
+    "volumes",
+    "build",
+    "dist",
+    ".git",
+    "generated",
+    "grafonnet-lib",
+    "prettier-config",
+    "lint-config",
+    "cache",
+    "artifacts",
+    "typechain",
+    "binaryen",
+    "system-contracts",
+    "artifacts-zk",
+    "cache-zk",
+    // Ignore directories with OZ and forge submodules.
+    "contracts/l1-contracts/lib",
+];
+
+const IGNORED_FILES: [&str; 4] = [
+    "KeysWithPlonkVerifier.sol",
+    "TokenInit.sol",
+    ".tslintrc.js",
+    ".prettierrc.js",
+];
+
+const CONFIG_PATH: &str = "etc/lint-config";
+
+#[derive(Debug, Parser)]
+pub struct LintArgs {
+    #[clap(long, short = 'c')]
+    pub check: bool,
+    #[clap(long, short = 'e')]
+    pub extensions: Vec<Extension>,
+}
+
+#[derive(Debug, ValueEnum, EnumIter, strum::Display, PartialEq, Eq, Clone)]
+pub enum Extension {
+    #[strum(serialize = "rs")]
+    Rs,
+    #[strum(serialize = "md")]
+    Md,
+    #[strum(serialize = "sol")]
+    Sol,
+    #[strum(serialize = "js")]
+    Js,
+    #[strum(serialize = "ts")]
+    Ts,
+}
+
+pub fn run(shell: &Shell, args: LintArgs) -> anyhow::Result<()> {
+    let extensions = if args.extensions.is_empty() {
+        vec![
+            Extension::Rs,
+            Extension::Md,
+            Extension::Sol,
+            Extension::Js,
+            Extension::Ts,
+        ]
+    } else {
+        args.extensions.clone()
+    };
+
+    logger::info(msg_running_linters_for_files(&extensions));
+
+    let ecosystem = EcosystemConfig::from_file(shell)?;
+
+    for extension in extensions {
+        match extension {
+            Extension::Rs => lint_rs(shell, &ecosystem)?,
+            Extension::Sol => lint_contracts(shell, &ecosystem, args.check)?,
+            ext => lint(shell, &ecosystem, &ext, args.check)?,
+        }
+    }
+
+    Ok(())
+}
+
+fn lint_rs(shell: &Shell, ecosystem: &EcosystemConfig) -> anyhow::Result<()> {
+    let spinner = Spinner::new(&msg_running_linter_for_extension_spinner(&Extension::Rs));
+
+    let link_to_code = &ecosystem.link_to_code;
+    let lint_to_prover = &ecosystem.link_to_code.join("prover");
+    let link_to_toolbox = &ecosystem.link_to_code.join("zk_toolbox");
+    let paths = vec![link_to_code, lint_to_prover, link_to_toolbox];
+
+    for path in paths {
+        let _dir_guard = shell.push_dir(path);
+        Cmd::new(cmd!(
+            shell,
+            "cargo clippy --locked -- -D warnings -D unstable_features"
+        ))
+        .run()?;
+    }
+
+    spinner.finish();
+
+    Ok(())
+}
+
+fn get_linter(extension: &Extension) -> Vec<String> {
+    match extension {
+        Extension::Rs => vec!["cargo".to_string(), "clippy".to_string()],
+        Extension::Md => vec!["markdownlint".to_string()],
+        Extension::Sol => vec!["solhint".to_string()],
+        Extension::Js => vec!["eslint".to_string()],
+        Extension::Ts => vec!["eslint".to_string(), "--ext".to_string(), "ts".to_string()],
```

Comment:
should we check these prerequisites?

Reply:
It is all run by yarn:
`yarn eslint`

And the deps are in `package.json`